catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/SettingsRepository.java

```diff
@@ -0,0 +1,137 @@
+package org.openmetadata.catalog.jdbi3;
+
+import static org.openmetadata.catalog.util.EntityUtil.compareEventFilters;
+import static org.openmetadata.catalog.util.EntityUtil.compareFilters;
+
+import java.util.List;
+import javax.json.JsonPatch;
+import javax.json.JsonValue;
+import javax.ws.rs.core.Response;
+import lombok.extern.slf4j.Slf4j;
+import org.openmetadata.catalog.filter.EventFilter;
+import org.openmetadata.catalog.filter.Filters;
+import org.openmetadata.catalog.settings.Settings;
+import org.openmetadata.catalog.settings.SettingsType;
+import org.openmetadata.catalog.util.JsonUtils;
+import org.openmetadata.catalog.util.RestUtil;
+import org.openmetadata.catalog.util.ResultList;
+
+@Slf4j
+public class SettingsRepository {
+  private final CollectionDAO dao;
+
+  public SettingsRepository(CollectionDAO dao) {
+    this.dao = dao;
+  }
+
+  public ResultList<Settings> listAllConfigs() {
+    List<Settings> settingsList = null;
+    try {
+      settingsList = dao.getSettingsDAO().getAllConfig();
+    } catch (Exception ex) {
+      LOG.error("Error while trying fetch all Settings " + ex.getMessage());
+    }
+    int count = 0;
+    if (settingsList != null) {
+      count = settingsList.size();
+    }
+    return new ResultList<>(settingsList, null, null, count);
+  }
+
+  public Settings getConfigWithKey(String key) {
+    Settings settings = null;
+    try {
+      settings = dao.getSettingsDAO().getConfigWithKey(key);
+    } catch (Exception ex) {
+      LOG.error("Error while trying fetch Settings " + ex.getMessage());
+    }
+    return settings;
+  }
+
+  public Response createOrUpdate(Settings setting) {
+    Settings oldValue = getConfigWithKey(setting.getConfigType().toString());
+    try {
+      updateSetting(setting);
+    } catch (Exception ex) {
+      LOG.error("Failed to Update Settings" + ex.getMessage());
+      return Response.status(500, "Internal Server Error. Reason :" + ex.getMessage()).build();
+    }
+    if (oldValue == null) {
+      return (new RestUtil.PutResponse<>(Response.Status.CREATED, setting, RestUtil.ENTITY_CREATED)).toResponse();
+    } else {
+      return (new RestUtil.PutResponse<>(Response.Status.OK, setting, RestUtil.ENTITY_UPDATED)).toResponse();
+    }
+  }
+
+  public Response updateEntityFilter(String entityType, List<Filters> filters) {
```

Comment:
You have FilterUtil , why don't we use that to have all the code related to EntityFilter in one place